plugin/kubernetes/kubernetes.go

```diff
@@ -291,6 +299,29 @@ func (k *Kubernetes) Records(state request.Request, exact bool) ([]msg.Service,
 	return services, err
 }
 
+// ServiceFQDN returns the k8s cluster dns spec service FQDN for the service (or endpoint) object
```

Comment:
In another PR we should update all the places we do this to use these functions.